tools/nixpkgs-hammer

```diff
@@ -17,17 +49,23 @@ def escape_nix_string(val: str) -> str:
 
 def nix_eval(expr: str) -> str:
     return subprocess.check_output(
-        [NIX_INSTANTIATE_PATH, '--eval', '-E', expr],
+        ['nix-instantiate', '--eval', '-E', expr],
         encoding='utf-8',
     )
 
 
-def nix_eval_json(expr: str, as_string=False):
-    result = subprocess.check_output(
-        [NIX_INSTANTIATE_PATH, '--strict', '--json', '--eval', '-E', expr],
-        encoding='utf-8',
-    )
-    return result if as_string else json.loads(result)
+def nix_eval_json(expr: str):
+    try:
+        result = subprocess.check_output(
+            ['nix-instantiate', '--strict', '--json', '--eval', '--show-trace', '-E', expr],
+            encoding='utf-8',
+        )
+    except subprocess.CalledProcessError as e:
```

Comment:
What use cases does this fail? Maybe we should fix it in the Nix expression. Either way this change looks unrelated so it should be split into a separate commit.

Reply:
It was only failing when I was trying to write the overlays (previous PR), and while they were in some intermediate state of not-yet-working.